testsuite/testsuite_impl/my_advanced_test/my_advanced_test_.go

```diff
@@ -0,0 +1,528 @@
+	}
+
+	return nil
+}
+
+// ====================================================================================================
+//                                       Private helper functions
+// ====================================================================================================
+func startEthBootnode(networkCtx *networks.NetworkContext) (string, error) {
+	containerCreationConfig, runConfigFunc := getEthereumServiceConfigurationsForBootnode()
+
+	_, hostPortBindings, err := networkCtx.AddService(bootnodeID, containerCreationConfig, runConfigFunc)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred adding the Ethereum Bootnode")
+	}
+
+	logrus.Infof("Added Ethereum Bootnode service with host port bindings: %+v ", hostPortBindings)
+
+	serviceCtx, err := networkCtx.GetServiceContext(bootnodeID)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred getting the Ethereum Bootnode service context")
+	}
+	logrus.Infof("Got service context for Ethereum Bootnode service '%v' and IP address '%v", serviceCtx.GetServiceID(), serviceCtx.GetIPAddress())
+
+	if err := waitForStartup(serviceCtx.GetIPAddress(), gethTimeBetweenIsAvailablePolls, gethMaxIsAvailablePolls); err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred waiting for service with ID '%v' to start", bootnodeID)
+	}
+
+	exitCode, logOutput, err := serviceCtx.ExecCommand([]string{
+		"/bin/sh",
+		"-c",
+		fmt.Sprintf("geth attach data/geth.ipc --exec admin.nodeInfo.enr"),
+	})
+	if err != nil {
+		return "", stacktrace.Propagate(err,"Executing command returned an error with logs: %+v", string(*logOutput))
+	}
+	if exitCode != execCommandSuccessExitCode {
+		return "", stacktrace.NewError("Executing command returned an failing exit code with logs: %+v", string(*logOutput))
+	}
+
+	enr := string(*logOutput)
+
+	return enr, nil
+}
+
+func starEthNodeByBootnode(networkCtx *networks.NetworkContext, serviceID services.ServiceID, bootnodeEnr string, nodesEnode []string) (string, error) {
+	containerCreationConfig, runConfigFunc := getEthereumServiceConfigurationsForNode(bootnodeEnr)
+	_, hostPortBindings, err := networkCtx.AddService(serviceID, containerCreationConfig, runConfigFunc)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred adding the Ethereum Node %v", serviceID)
+	}
+
+	logrus.Infof("Added Ethereum Node %v service with host port bindings: %+v ", serviceID, hostPortBindings)
+
+	serviceCtx, err := networkCtx.GetServiceContext(serviceID)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred getting the Ethereum Node %v service context", serviceID)
+	}
+	logrus.Infof("Got service context for Ethereum Node '%v' and IP address '%v'", serviceCtx.GetServiceID(), serviceCtx.GetIPAddress())
+
+	if err := waitForStartup(serviceCtx.GetIPAddress(), gethTimeBetweenIsAvailablePolls, gethMaxIsAvailablePolls); err != nil {
```

Comment:
I didn't use the method `WaitForEndpointAvailability` from Kurt Core because it only executes HTTP Get requests and it needs the HTTP Post method for RPC calls.